warpscript/utapi/createSnapshot.mc2

```diff
@@ -40,15 +40,54 @@
     'utapi.checkpoint' 'checkpoint_class' STORE
     'utapi.repair.correction' 'correction_class' STORE
 
+    // will read the key `max_snapshot_size@utapi/createSnapshot` from warp 10 config
+    'max_snapshot_size' 'P6H' MACROCONFIGDEFAULT DURATION 'max_snapshot_size' STORE
+
     // Fetch latest master snapshot
     $read_token $master_snapshot_class $filterLabels $endTimestamp @utapi/fetchFirstRecordBefore
 
+
     FIRSTTICK
     // If we found a snapshot, increment its timestamp so we start at the tick immediatly after
     <% DUP 0 > %>
     <%  1 + %> IFT
     'masterSnapshotTimestamp' STORE // Grab our ending timestamp from the last master snapshot (0 if no snapshots)
 
+    // If snapshot is longer than 6 hours change endTimestamp to be masterSnapshotTimestamp + 6 hours
+    <% $endTimestamp $masterSnapshotTimestamp - $max_snapshot_size > %>
+    <%
+      // If there are no preceding snapshots
+      <% $masterSnapshotTimestamp 0 == %>
+      <%
+        // Find the oldest checkpoint
+        // 'No previous snapshots found, detecting start of checkpoints.' LOGMSG
+        $auth_info ->JSON 
+        { 'class' $checkpoint_class 'labels' $filterLabels } ->JSON
+        @utapi/findOldestRecord 'oldestTimestamp' STORE
+        // If no checkpoints are found 
+        <% $oldestTimestamp -1 == %>
+        <%
+          // 'No checkpoints found. Looking for corrections' LOGMSG
+          $auth_info ->JSON 
+          { 'class' $correction_class 'labels' $filterLabels } ->JSON
+          @utapi/findOldestRecord 'oldestTimestamp' STORE
+          <% $oldestTimestamp -1 == %>
+          <%
+            // 'No corrections or checkpoints found. Nothing to snapshot' LOGMSG
+            STOP
+          %> IFT
+        %>  IFT
+        <% $oldestTimestamp $max_snapshot_size + $endTimestamp < %>
+        <%
+          $oldestTimestamp $max_snapshot_size + 'endTimestamp' STORE
+        %> IFT
+      %> 
+      <%
+        $masterSnapshotTimestamp $max_snapshot_size + 'endTimestamp' STORE 
```

Comment:
Not sure if I get this right, if here we are in the "else" clause of this check at https://github.com/scality/utapi/pull/1209/files#diff-0547afee6eaac48bd3f26855e5b9adbcb9bd4e056ba6d04ef792027da0a93b0aR57:

```
<% $endTimestamp $masterSnapshotTimestamp - $max_snapshot_size > %>
```

Wouldn't this mean we are requesting a range less than 6 hours since the last master snapshot, hence keep the end timestamp as requested, instead of using `$max_snapshot_size` and storing it in `endTimestamp`?

Reply:
The nesting is a bit confusing, but this is the else for `<% $masterSnapshotTimestamp 0 == %>`
Meaning we are in the "We have a previous snapshot" case

Reply:
ah indeed